domains/authorization/Authorization.API/API.IntegrationTests/API.IntegrationTests/API/AccessTokenTests.cs

```diff
@@ -0,0 +1,104 @@
+using API.Models;
+using API.UnitTests;
+using API.ValueObjects;
+using FluentAssertions;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.IdentityModel.Tokens;
+
+namespace API.IntegrationTests.API;
+
+[Collection(IntegrationTestCollection.CollectionName)]
+public class AccessTokenTests
+{
+    private readonly Api _api;
+    private readonly IntegrationTestFixture _integrationTestFixture;
+    private readonly DbContextOptions<ApplicationDbContext> _options;
+
+    public AccessTokenTests(IntegrationTestFixture integrationTestFixture)
+    {
+        _integrationTestFixture = integrationTestFixture;
+        var connectionString = integrationTestFixture.WebAppFactory.ConnectionString;
+        _options = new DbContextOptionsBuilder<ApplicationDbContext>().UseNpgsql(connectionString).Options;
+        _api = integrationTestFixture.WebAppFactory.CreateApi();
+    }
+
+    [Fact]
+    public async Task GivenConsent_WhenDeleted_ThenAccessTokenShouldNotContainOrganizationId()
+    {
+        // Step 1: Seed database with a user, an organization, and a client
+        var user = Any.User();
+        var organization = Any.Organization();
+        var client = Client.Create(new IdpClientId(Guid.NewGuid()), new("Loz"), ClientType.Internal, "https://localhost:5001");
+        var consent = Consent.Create(organization, client, DateTimeOffset.UtcNow);
+
+        await using var dbContext = new ApplicationDbContext(_options);
+        await dbContext.Users.AddAsync(user);
+        await dbContext.Organizations.AddAsync(organization);
+        await dbContext.Clients.AddAsync(client);
+        await dbContext.Consents.AddAsync(consent);
+        var affiliation = Affiliation.Create(user, organization);
+        await dbContext.Affiliations.AddAsync(affiliation);
+        await dbContext.SaveChangesAsync();
+
+        // Step 2: Generate and validate access token before deletion
+        var accessTokenBefore = GenerateToken(client.IdpClientId.Value.ToString(), user.Name.Value, organization.Id.ToString(), "user", organization.Tin.Value);
+        var orgIdsBefore = GetOrgIdsFromAccessToken(accessTokenBefore);
+        orgIdsBefore.Should().Contain(organization.Id.ToString());
+
+        // Step 3: Delete the consent
+        var userClient = _integrationTestFixture.WebAppFactory.CreateApi(sub: user.IdpUserId.Value.ToString(), orgCvr: organization.Tin.Value);
+        var deleteResponse = await userClient.DeleteConsent(client.Id, organization.Id);
+        deleteResponse.Should().Be204NoContent();
+
+        // Step 4: Generate and validate access token after deletion
+        var accessTokenAfter = GenerateToken(client.IdpClientId.Value.ToString(), user.Name.Value, "", "user", organization.Tin.Value);
+        var orgIdsAfter = GetOrgIdsFromAccessToken(accessTokenAfter);
+        orgIdsAfter.Should().NotContain(organization.Id.ToString());
+    }
+
+    private string GenerateToken(string sub, string name, string orgIds, string subType, string orgCvr)
```

Comment:
No, we don't generate tokens in this subsystem. I don't believe this is correct, we should not be able to generate correct tokens in this sub system, this is clearly a mistake.